src/pngmaker/new_validated_challenge.py

```diff
@@ -0,0 +1,85 @@
+from os import getenv
+from PIL import Image,ImageDraw,ImageFont
+
+class NewValidatedChallenge() :
+    """
+    Class that creates an image when initialized
+    """
+
+    def __init__(self,user,challenge,order) -> None:
+        #load PYTHONPATH to get the assets path
+        if getenv("PYTHONPATH") is not None :
+            self.assets_path = getenv("PYTHONPATH")+"/assets"
+        else :
+            raise Exception("PYTHONPATH env variable not set")
+        #take the background picture for base
+        self.image = Image.open(self.assets_path+"/bg_dark.jpg")
+        # fill the picture
+        self.make_title()
+        self.make_profile(user)
+        self.make_challenge(challenge)
+        self.make_challenge_category(challenge)
+        self.make_order(order)
```

Comment:
I did that just to split the different parts of the image making process. I don't think it's a good idea to put it in a `__call__` function because then to get the image you would have to instantiate the class, then call it to "make" the png and then use `object.image`. Instead putting everything in `__init__` is a bit ugly i agree but when you instantiate the object the image is made and just accessed by `object.image` when you need it.

Reply:
Let's go with it then!